newt/repo/repo.go

```diff
@@ -219,12 +219,19 @@ func (r *Repo) checkExists() bool {
 }
 
 func (r *Repo) updateRepo(branchName string) error {
-	dl := r.downloader
-	err := dl.UpdateRepo(r.Path(), branchName)
+	err := r.downloader.UpdateRepo(r.Path(), branchName)
 	if err != nil {
-		return util.FmtNewtError("Error updating \"%s\": %s",
-			r.Name(), err.Error())
+		// If the update failed because the repo directory has been deleted,
+		// clone the repo again.
+		if util.IsNotExist(err) {
+			err = r.downloadRepo(branchName)
+		}
 	}
+	if err != nil {
```

Comment:
Lines 229-230 could be removed.

Reply:
Just removing those lines slightly changes the behaviour... following the path of `err != nil` in line 223 might run `downloadRepo` and return `err == nil` in which case it would end up running `util.FmtNewError`...

Reply:
True. My mistake in comment :) I meant line 229 only; no need to check err == nil twice.